### PR TITLE
OCSADV-355: SDSS filters must default to AB mag system

### PR DESCRIPTION
**Pull request by cquiroz**
Update to the target editor, when a magnitude band belonging to the AB system is selected, set the magnitude system to AB. Else, select the Vega system
A few code simplifications and warnings removal are also included

### CODE REVIEW

File: bundle/edu.gemini.shared.skyobject/src/main/java/edu/gemini/shared/skyobject/Magnitude.java [hunk 1]
```diff
@@ -45,6 +48,18 @@
...
+
+        static {
+            List<Band> bands = new ArrayList<>();
+            bands.add(u);
+            bands.add(g);
+            bands.add(r);
+            bands.add(i);
+            bands.add(z);
+            AB_BANDS = Collections.unmodifiableList(bands);
+        }
```
Comment (swalker2m) [inline bundle/edu.gemini.shared.skyobject/src/main/java/edu/gemini/shared/skyobject/Magnitude.java]: What about putting the default system as a field of the band itself?  That would force you to think about it whenever you add a new band.

**Comment by swalker2m** [comment]
This raises a question about AGS.  We treat r, R, and UC as interchangeable for the purpose of determining and comparing magnitudes.  If the units, as it were, aren't even the same though  can we really do that?

File: bundle/edu.gemini.shared.skyobject/src/main/java/edu/gemini/shared/skyobject/Magnitude.java (hunk 1 above)
Comment (krumpi) [inline bundle/edu.gemini.shared.skyobject/src/main/java/edu/gemini/shared/skyobject/Magnitude.java]: I was thinking also if we should let people change the system if bands already have one

Comment (cquiroz) [inline bundle/edu.gemini.shared.skyobject/src/main/java/edu/gemini/shared/skyobject/Magnitude.java]: I was thinking also if we should let people change the system if bands already have one

**Comment by cquiroz** [comment]
So a magnitude 7 in AB is not the same as a 7 in VEGA? If so you are probably right.

File: bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/MagnitudeEditor.java [hunk 2]
```diff
@@ -98,11 +97,18 @@ public static synchronized String formatBrightness(Magnitude mag) {
...
         // new magnitude band.
         private final ActionListener changeBandAction = new ActionListener() {
+
             @Override public void actionPerformed(ActionEvent e) {
                 final Magnitude.Band newBand = (Magnitude.Band) cb.getSelectedItem();
                 if (newBand == null) return;
                 if (newBand == MagEditRow.this.band) return;
                 changeBand(MagEditRow.this.band, newBand);
+                // OCSADV-355 If the bands is one on the AB list, switch the magnitude system to AB or else to VEGA
+                if (Magnitude.Band.AB_BANDS.stream().anyMatch(b -> b.equals(newBand))) {
```
Comment (swalker2m) [inline bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/MagnitudeEditor.java]: Hadn't notice Java `Stream` before.  Nice.

**Comment by cquiroz** [comment]
I'll merge to let @swalker2m continue

**Pull request merged**